internal/agent/device/status/status_test.go

```diff
@@ -0,0 +1,52 @@
+package status
```

Comment:
I suspect the benchmark sneaked in and is unrelated

Reply:
But very glad to see benchmarks :)

Reply:
Yeah it was related in the sense that I used it to validate the leak was not in status collection. We can use the benchmark in the future to ensure allocations and serialization costs are reasonable.